deployment/cdk/opensearch-service-migration/lib/historical-capture-stack.ts

```diff
@@ -1,62 +1,97 @@
+    readonly sourceEndpoint: string,
     readonly targetEndpoint: string
 }
 
-/**
- * This stack was a short exploratory task into having a deployable Logstash ECS cluster for historical data migration.
- * NOTE: It should only be used for development purposes in its current state
- */
 export class HistoricalCaptureStack extends Stack {
 
     constructor(scope: Construct, id: string, props: historicalCaptureStackProps) {
         super(scope, id, props);
 
+        // ECS Cluster in VPC
         const ecsCluster = new Cluster(this, "ecsHistoricalCaptureCluster", {
             vpc: props.vpc
         });
 
+        // ECS Task Definition
         const historicalCaptureFargateTask = new FargateTaskDefinition(this, "historicalCaptureFargateTask", {
             memoryLimitMiB: 2048,
             cpu: 512
         });
 
-        let logstashConfigData: string = readFileSync(props.logstashConfigFilePath, 'utf8');
-        if (props.sourceEndpoint) {
-            logstashConfigData = logstashConfigData.replace("<SOURCE_CLUSTER_HOST>", props.sourceEndpoint)
-        }
-        logstashConfigData = logstashConfigData.replace("<TARGET_CLUSTER_HOST>", props.targetEndpoint + ":80")
-        // Temporary measure to allow multi-line env variable
-        logstashConfigData = logstashConfigData.replace(/(\n)/g, "PUT_LINE")
         // Create Historical Capture Container
-        const historicalCaptureImage = new DockerImageAsset(this, "historicalCaptureImage", {
-            directory: join(__dirname, "../../..", "docker/logstash-setup")
-        });
-
         const historicalCaptureContainer = historicalCaptureFargateTask.addContainer("historicalCaptureContainer", {
-            image: ContainerImage.fromDockerImageAsset(historicalCaptureImage),
+            image: ContainerImage.fromAsset(join(__dirname, "../../../..", "FetchMigration")),
             // Add in region and stage
-            containerName: "logstash",
-            environment: {"LOGSTASH_CONFIG": '' + logstashConfigData},
-            logging: LogDrivers.awsLogs({ streamPrefix: 'logstash-lg', logRetention: 30 })
+            containerName: "fetch-migration",
+            logging: LogDrivers.awsLogs({ streamPrefix: 'fetch-migration-lg', logRetention: 30 })
+        });
+
+        // Create DP pipeline config from template file
+        let dpPipelineData: string = readFileSync(props.dpPipelineTemplatePath, 'utf8');
+        dpPipelineData = dpPipelineData.replace("<SOURCE_CLUSTER_HOST>", props.sourceEndpoint);
+        dpPipelineData = dpPipelineData.replace("<TARGET_CLUSTER_HOST>", props.targetEndpoint);
+        // Base64 encode
+        let encodedPipeline = Buffer.from(dpPipelineData).toString("base64");
+
+        // Create secret using Secrets Manager
+        const dpPipelineConfigSecret = new SMSecret(this, "dpPipelineConfigSecret", {
+            secretName: `${historicalCaptureFargateTask.family}-${historicalCaptureContainer.containerName}-pipelineConfig`,
```

Comment:
I don't think we have to address this right now (as I believe there are a couple other spots in the IaC as well), but worth calling out that if a user deployed multiple instances of our migration solution in the same region (e.g. if they were migrating multiple target clusters separately) I would expect this name to cause a collision